packages/cozy-authentication/MobileRouter.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react'
+import { Router, withRouter } from 'react-router'
+import Proptypes from 'prop-types'
+import Authentication from './src/Authentication'
+import Revoked from './src/Revoked'
+import {
+  readState,
+  secretExchange,
+  getAccessToken,
+  readSecret,
+  clearSecret,
+  clearState,
+  checkIfOnboardingLogin,
+  addProtocolToURL
+} from './src/utils/onboarding'
+import {
+  onboardingInformationsPropTypes,
+  onboardingPropTypes
+} from './OnboardingPropTypes'
+export class MobileRouter extends Component {
+  async doOnboardingLogin(receivedState, code, instanceDomain, history) {
+    const localState = await readState()
+    const localSecret = await readSecret()
+
+    try {
+      if (localState !== receivedState) {
+        throw new Error('States are not equals')
+      }
+
+      const clientInfo = await secretExchange(
+        localSecret,
+        instanceDomain,
+        this.props.client
+      )
+
+      const {
+        onboarding_secret,
+        onboarding_state,
+        client_id,
+        client_secret
+      } = clientInfo
+
+      if (
+        !(localSecret === onboarding_secret && localState === onboarding_state)
+      )
+        throw new Error('exchanged informations are not good')
+
+      const token = await getAccessToken(
```

Comment:
Nit: I tend to use get for synchronous functions and fetch for asynchronous ones.